File: examples/fibonacci.py
memo = {}


def fibo(n):
    if n <= 2:
        return 1

    if n in memo:
        return memo[n]

    memo[n] = fibo(n - 2) + fibo(n - 1)
    return memo[n]


for i in range(1, 25 + 1):
    print(i, fibo(i))

File: examples/fibonacci.js
const mem = {};

const fibo = (N) => {
  if (N <= 2) return 1;

  if (mem[N]) return mem[N];

  mem[N] = fibo(N - 2) + fibo(N - 1);
  return mem[N];
};

const result = fibo(10);
console.log(result);
